opensearch/user_data.yaml.tpl

```diff
@@ -39,6 +39,56 @@ write_files:
     permissions: "0400"
     content: |
       ${indent(6, tls.admin_key)}
+%{ if try(snapshot_repository.ca_cert, "") != "" ~}
+  - path: /etc/opensearch/snapshot-repository/ca.crt
+    owner: root:root
+    permissions: "0400"
+    content: |
+      ${indent(6, snapshot_repository.ca_cert)}
+  - path: /usr/local/bin/import_snapshot_repository_ca
+    owner: root:root
+    permissions: "0555"
+    content: |
+      #!/bin/bash
+      set -euo pipefail
+
+      CA_FILE=/etc/opensearch/snapshot-repository/ca.crt
+      CACERTS="/opt/opensearch/jdk/lib/security/cacerts"
+      OS_KEYSTORE="/etc/opensearch/configuration/opensearch.keystore"
+
+      if [ ! -s "$CA_FILE" ]; then
+        exit 0
+      fi
```

Comment:
Given your template if, this is an impossibility. Might as well remove that if statement here and make the script a bit simpler.